internal/controller/replication.storage/status.go

```diff
@@ -140,20 +157,23 @@ func setDemotedCondition(conditions *[]metav1.Condition, observedGeneration int6
 }
 
 // sets conditions when volume demotion was failed.
-func setFailedDemotionCondition(conditions *[]metav1.Condition, observedGeneration int64) {
+func setFailedDemotionCondition(conditions *[]metav1.Condition, observedGeneration int64, message, messageFromDriver, dataSource string) {
```

Comment:
For consistency, dataSource should be before message and messageFromDriver. It is easier to work with code that have consistent structure. If a group of function have common arguments, they should be in the same order.

To make the other arguments more clear, maybe:
- completedMessage
- degradedDetailedMessage?

With this is is very clear which messages belong to which condition and it is clear that completedMessage is not the same content as degradedDetails. One is message and the other is something longer and more detailed.

Same for the other functions getting datasource, message, messageFromDriver.